src/engine/server/server.cpp

```diff
@@ -1214,7 +1214,7 @@ void CServer::SendMap(int ClientId)
 		Msg.AddRaw(&m_aCurrentMapSha256[MapType].data, sizeof(m_aCurrentMapSha256[MapType].data));
 		Msg.AddInt(m_aCurrentMapCrc[MapType]);
 		Msg.AddInt(m_aCurrentMapSize[MapType]);
-		Msg.AddString("", 0); // HTTPS map download URL
+		Msg.AddString(m_MapDownloadUrl[MapType].c_str(), 0);
```

Comment:
```suggestion
		if(!m_MapDownloadUrl[MapType].empty())
		{
			Msg.AddString(m_MapDownloadUrl[MapType].c_str(), 0);
		}
		else
		{
			Msg.AddString("", 0);
		}
```
I think a conditional would be clearer here, emphasizing that this is an optional part of the message. Skip this if you disagree.

Reply:
As you wish.